### PR TITLE
Refactor: Replace class to hooks on object list component

### PR DESCRIPTION
**Pull request by jane-rose**

### CODE REVIEW

File: src/app/widgets/CncLaserList/ObjectList/ObjectListBox.jsx [hunk 1]
```diff
@@ -10,152 +11,108 @@ import modal from '../../../lib/modal';
...
 
-class ObjectListBox extends PureComponent {
-    static propTypes = {
-        setTitle: PropTypes.func.isRequired,
-        selectTargetModel: PropTypes.func.isRequired,
-        hideSelectedModel: PropTypes.func.isRequired,
-        showSelectedModel: PropTypes.func.isRequired,
+const ObjectListBox = (props) => {
+    const selectedModelArray = useSelector(state => state[props.headType].modelGroup && state[props.headType].modelGroup.selectedModelArray);
+    const models = useSelector(state => state[props.headType].modelGroup && state[props.headType].modelGroup.models);
```
Comment (parachvte) [inline src/app/widgets/CncLaserList/ObjectList/ObjectListBox.jsx:16]: Check properties correctly.

Comment (jane-rose) [reply src/app/widgets/CncLaserList/ObjectList/ObjectListBox.jsx]: Updated @parachvte

File: src/app/lib/redux-select.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { createSelector } from 'reselect';
+
+export const reduxSelector = (allState, attributeArray) => {
```
Comment (parachvte) [inline src/app/lib/redux-select.js:3]: getAttribute(object, attributePath) {

}

File: src/app/flux/editor/actions-process.js [hunk 3]
```diff
@@ -154,7 +165,7 @@ export const processActions = {
 
     saveToolPath: (headType, toolPath) => (dispatch, getState) => {
         const { toolPathGroup, materials, autoPreviewEnabled } = getState()[headType];
-
+        console.log('saveToolPath', toolPath);
```
Comment (parachvte) [inline src/app/flux/editor/actions-process.js:168]: Remember to remove log.

File: src/server/lib/ToolPathGenerator/CncReliefToolPathGenerator.js [hunk 4]
```diff
@@ -137,6 +137,7 @@ export default class CncReliefToolPathGenerator extends EventEmitter {
         return Jimp
             .read(this.modelPath)
             .then(async img => {
+                console.log('_processImage', img, this.modelPath);
```
Comment (meiliang1123) [inline src/server/lib/ToolPathGenerator/CncReliefToolPathGenerator.js:140]: logs

**Review by meiliang1123 — CHANGES_REQUESTED** [review]

File: src/app/rehooks/widget-title.jsx [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { useEffect } from 'react';
+import i18n from '../lib/i18n';
+
+const useWidgetTitle = (setTitle, title, otherFunction) => {
```
Comment (parachvte) [inline src/app/rehooks/widget-title.jsx:4]: Not being used?

**Review by meiliang1123 — APPROVED** [review]

**Pull request merged**